compiler error when crypto using picotls with openssl 1.0.2

Hi,

Due to some reason, I had to compile codes on CentOS 7 which only has openssl-1.0.2k. `picotls` or `quicly` can both compile and works, but `ngtcp2` will run into error:

```
Making all in picotls
make[3]: Entering directory '/home/chiouss/ngtcp2/crypto/picotls'
  CC       picotls.o
picotls.c: In function ‘crypto_ptls_get_aead_max_encryption’:
picotls.c:85:20: error: ‘ptls_openssl_chacha20poly1305’ undeclared (first use in this function)
   if (cs->aead == &ptls_openssl_chacha20poly1305) {
                    ^
picotls.c:85:20: note: each undeclared identifier is reported only once for each function it appears in
picotls.c: In function ‘crypto_ptls_get_aead_max_decryption_failure’:
picotls.c:100:20: error: ‘ptls_openssl_chacha20poly1305’ undeclared (first use in this function)
   if (cs->aead == &ptls_openssl_chacha20poly1305) {
                    ^
picotls.c: In function ‘crypto_ptls_get_hp’:
picotls.c:118:20: error: ‘ptls_openssl_chacha20poly1305’ undeclared (first use in this function)
   if (cs->aead == &ptls_openssl_chacha20poly1305) {
                    ^
picotls.c:119:13: error: ‘ptls_openssl_chacha20’ undeclared (first use in this function)
     return &ptls_openssl_chacha20;
             ^
Makefile:517: recipe for target 'picotls.o' failed
make[3]: *** [picotls.o] Error 1
make[3]: Leaving directory '/home/chiouss/ngtcp2/crypto/picotls'
Makefile:439: recipe for target 'all-recursive' failed
```

I think we can add `#if PTLS_OPENSSL_HAVE_CHACHA20_POLY1305` there?

--- a/crypto/picotls/picotls.c
+++ b/crypto/picotls/picotls.c
@@ -82,9 +82,11 @@ static uint64_t crypto_ptls_get_aead_max_encryption(ptls_t *ptls) {
     return NGTCP2_CRYPTO_MAX_ENCRYPTION_AES_GCM;
   }
 
+#if PTLS_OPENSSL_HAVE_CHACHA20_POLY1305
   if (cs->aead == &ptls_openssl_chacha20poly1305) {
     return NGTCP2_CRYPTO_MAX_ENCRYPTION_CHACHA20_POLY1305;
   }
+#endif
 
   return 0;
 }
@@ -97,9 +99,11 @@ static uint64_t crypto_ptls_get_aead_max_decryption_failure(ptls_t *ptls) {
     return NGTCP2_CRYPTO_MAX_DECRYPTION_FAILURE_AES_GCM;
   }
 
+#if PTLS_OPENSSL_HAVE_CHACHA20_POLY1305
   if (cs->aead == &ptls_openssl_chacha20poly1305) {
     return NGTCP2_CRYPTO_MAX_DECRYPTION_FAILURE_CHACHA20_POLY1305;
   }
+#endif
 
   return 0;
 }
@@ -115,9 +119,11 @@ static const ptls_cipher_algorithm_t *crypto_ptls_get_hp(ptls_t *ptls) {
     return &ptls_openssl_aes256ctr;
   }
 
+#if PTLS_OPENSSL_HAVE_CHACHA20_POLY1305
   if (cs->aead == &ptls_openssl_chacha20poly1305) {
     return &ptls_openssl_chacha20;
   }
+#endif
 
   return NULL;
 }

